File: solutions2020/PRB14/nkadaba627@student.fuhsd.org.java
import java.util.Arrays;
import java.util.Scanner;

public class PRB14 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String s = sc.nextLine();
		int n = Integer.parseInt(s);
		String[] raw = new String[n];
		for(int i = 0;i<n;i++){
			raw[i] = sc.nextLine();
		}
		sc.close();
		for(int i = 0;i<n;i++){
			System.out.println(work(raw[i]));
		}
	}

	public static int work(String s) {
		String[] raw = s.split(" ");
		int[] nums = new int[raw.length];
		for (int i = 0; i < raw.length; i++) {
			nums[i] = Integer.parseInt(raw[i]);
		}
		int a = 1;
		Arrays.sort(nums);
		 
        // Traverse the array and increment 'res' if arr[i] is
        // smaller than or equal to 'res'.
        for (int i = 0; i < nums.length && nums[i] <= a; i++)
            a = a + nums[i];
 
        return a;
		
	}
	
	
}

File: solutions2020/PRA3_1/nkadaba627@student.fuhsd.org.cpp
import java.util.Arrays;
import java.util.Scanner;

public class PRA3_1 {

	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int n = Integer.parseInt(s.nextLine());
		int[] ans = new int[n];
		for(int j = 0;j<n;j++) {
			String[] a = s.nextLine().split(" ");
			int k = Integer.parseInt(a[0]);
			int i = Integer.parseInt(a[1]);
			a = s.nextLine().split(" ");
			int[] nums = new int[i];
			for(int l = 0;l<i;l++) {
				nums[l] = Integer.parseInt(a[l]);
			}
			Arrays.sort(nums);
			int e = 0;
			int temp = 0;
			for(int q = i-1;q>0;q--) {
				if(temp==0) {
					temp = k - nums[q];
					e += temp+nums[q];
				} else {
					temp = temp - nums[q];
					e += temp+nums[q];
				}
			}
			ans[j] = e;
		}
		for(int p:ans) {
			System.out.println(p);
		}
	}

}
